src/main/java/com/chatsounds/ChatSoundsPlugin.java

```diff
@@ -87,82 +100,152 @@ protected void startUp()
+				}
+				break;
+
 			case CLAN_CHAT:
-				playSound(config.clanChat(), CS_CLAN);
+				if (shouldIgnorePlayer(clanIgnored, cleanName)) {
+					return;
+				}
+				playSound(config.clanChat(), CS_CLAN, config.clanVolume());
 				break;
+
 			case CLAN_MESSAGE:
-				if (!msg.equals(CS_CLAN_MSG)) {
-					playSound(config.clanBroadcast(), CS_CLAN_BROADCAST);
+				if (shouldAlertOnJoinOrLeft(msg, config.clanIgnoreJoinLeave()) && !msg.equals(CS_CLAN_MSG)) {
+					playSound(config.clanBroadcast(), CS_CLAN_BROADCAST, config.clanVolume());
 				}
 				break;
-			case CLAN_GIM_CHAT:
-				playSound(config.gimChat(), CS_GIM);
+
+			case CLAN_GUEST_CHAT:
+				if (shouldIgnorePlayer(guestClanIgnored, cleanName)) {
+					return;
+				}
+				playSound(config.guestClanChat(), CS_CLAN_GUEST, config.guestClanVolume());
 				break;
-			case CLAN_GIM_MESSAGE:
-				if (!msg.equals(CS_GIM_MSG)) {
-					playSound(config.gimBroadcast(), CS_CLAN_BROADCAST);
+
+			case CLAN_GUEST_MESSAGE:
+				if (shouldAlertOnJoinOrLeft(msg, config.guestClanIgnoreJoinLeave()) &&
+						!msg.startsWith(CS_CLAN_GUEST_MSG_1) && !msg.endsWith(CS_CLAN_GUEST_MSG_2) &&
+						!msg.equals(CS_CLAN_GUEST_MSG_3)) {
+					playSound(config.guestClanBroadcast(), CS_CLAN_GUEST_BROADCAST, config.guestClanVolume());
 				}
 				break;
-			case CLAN_GUEST_CHAT:
-				playSound(config.clanGuestChat(), CS_CLAN_GUEST);
+
+			case CLAN_GIM_CHAT:
+				if (shouldIgnorePlayer(gimIgnored, cleanName)) {
+					return;
+				}
+				playSound(config.gimChat(), CS_GIM, config.groupIronVolume());
 				break;
-			case CLAN_GUEST_MESSAGE:
-				Matcher m = CS_CLAN_GUEST_PATTERN.matcher(msg);
-				if (m.find() || msg.equals(CS_CLAN_GUEST_MSG))
-				{
-					break;
+
+			case CLAN_GIM_MESSAGE:
+				if (!msg.equals(CS_GIM_MSG)) {
+					playSound(config.gimBroadcast(), CS_GIM_BROADCAST, config.groupIronVolume());
 				}
-				playSound(config.clanGuestSystemMessage(), CS_CLAN_GUEST_SYSTEM);
 				break;
+
 			case TRADEREQ:
-				playSound(config.tradeRequest(), CS_TRADE_REQUEST);
+				if (shouldIgnorePlayer(tradeIgnored, cleanName)) {
+					return;
+				}
+				playSound(config.tradeRequest(), CS_TRADE_REQUEST, config.tradeVolume());
 				break;
+
 			case CHALREQ_TRADE:
-				playSound(config.duelRequest(), CS_DUEL_REQUEST);
+				if (shouldIgnorePlayer(duelIgnored, cleanName)) {
+					return;
+				}
+				playSound(config.duelRequest(), CS_DUEL_REQUEST, config.duelVolume());
 				break;
 		}
 	}
 
 	@Subscribe
 	public void onConfigChanged(ConfigChanged configChanged)
 	{
-		if (configChanged.getGroup().equals(ChatSoundsConfig.GROUP) && configChanged.getKey().equals(ChatSoundsConfig.PLAYER_IGNORE_LIST))
+		if (configChanged.getGroup().equals("chatSounds"))
 		{
 			updateLists();
 		}
 	}
 
+	// Returns true if the message is from an ignored player in the chat's type.
+	private boolean shouldIgnorePlayer(List<String> ignoreList, String name) {
+        return ignoreList.contains(name.toLowerCase());
+    }
```

Comment:
Pretty sure the bug is here where I forgot to make the ignore list names lowercase before comparison to the lowercase player name. Pushing a change to this now.